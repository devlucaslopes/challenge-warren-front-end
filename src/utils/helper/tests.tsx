import React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'

import GlobalStyles from '../../styles/global'
import theme from '../../styles/theme'
import {
  HandleFilterProps,
  TransactionContext,
  TransactionContextProps
} from '../../contexts/TransactionContext'
import { ITransaction } from '../../models/Transaction'

type CustomContextProps = {
  transactions?: ITransaction[]
  filteredTransactions?: ITransaction[]
  modalIsOpen?: boolean
  handleFilter?: (data: HandleFilterProps) => void
  toggleModal?: () => void
  findTransactionById?: (id: string) => ITransaction | undefined
}

export const mockContext: TransactionContextProps = {
  transactions: [],
  filteredTransactions: [],
  modalIsOpen: false,
  toggleModal: jest.fn(),
  findTransactionById: jest.fn(),
  handleFilter: jest.fn()
}

export const renderWithTheme = (
  children: React.ReactNode,
  customContext: CustomContextProps
) => {
  return render(
    <TransactionContext.Provider value={{ ...mockContext, ...customContext }}>
      <ThemeProvider theme={theme}>
        <GlobalStyles {...theme} />
        {children}
      </ThemeProvider>
    </TransactionContext.Provider>
  )
}
