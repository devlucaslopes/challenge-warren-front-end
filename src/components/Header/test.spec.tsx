import React from 'react'

import { Header } from './'
import { renderWithTheme } from '../../utils/helper/tests'

describe('<Header />', () => {
  it('should render header correctly', () => {
    const { container } = renderWithTheme(<Header />)

    expect(container).toMatchSnapshot()
  })
})
