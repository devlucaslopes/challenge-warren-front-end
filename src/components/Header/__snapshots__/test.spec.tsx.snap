// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render header correctly 1`] = `
<div>
  <header
    class="sc-bdvvtL aAvJc"
    data-testid="header"
  >
    <div
      class="sc-gsDKAQ iToyxD"
    >
      <svg
        class="sc-dkPtRN jSVYFw"
        fill="none"
        height="30"
        viewBox="0 0 92 24"
        width="92"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M45.407 7.688h-2.16c-.3 0-.48.12-.56.421l-1.463 5.48-1.695-5.48c-.08-.279-.277-.418-.554-.42v-.001H37.41c-.277.003-.475.142-.555.421l-1.694 5.48-1.464-5.48c-.08-.3-.26-.421-.56-.421h-2.16c-.36 0-.519.22-.42.562l2.88 9.053c.08.28.3.421.58.421h1.919c.28 0 .5-.14.58-.421l1.676-5.273 1.677 5.273c.08.28.3.421.58.421h1.919c.28 0 .5-.14.58-.421l2.879-9.053c.1-.341-.06-.562-.42-.562Zm5.803 7.49c-1.324 0-2.396-1.109-2.396-2.476s1.072-2.475 2.396-2.475c1.323 0 2.395 1.108 2.395 2.475s-1.072 2.476-2.395 2.476Zm4.892-7.49h-1.997c-.32 0-.5.18-.5.501v.447c-.819-.767-1.873-1.23-3.024-1.23-2.608 0-4.722 2.372-4.722 5.296 0 2.925 2.114 5.296 4.722 5.296 1.15 0 2.205-.462 3.024-1.23v.448c0 .32.18.501.5.501h1.998c.32 0 .499-.18.499-.501V8.189c0-.321-.18-.501-.5-.501Zm35.106 3.673v5.861c0 .322-.18.502-.5.502h-2c-.32 0-.499-.18-.499-.502v-5.6c0-1.004-.58-1.445-1.34-1.445-.899 0-1.46.502-1.46 1.686v5.36c0 .32-.18.501-.499.501h-2c-.319 0-.5-.18-.5-.502V8.19c0-.32.181-.501.5-.501h2c.32 0 .5.18.5.502v.414c.47-.697 1.321-1.197 2.599-1.197 1.86 0 3.199 1.445 3.199 3.954Zm-12.803.261c-.31-1.142-1.191-1.546-2.073-1.546-1.124 0-1.902.545-2.21 1.546h4.283Zm2.985 1.084c0 .26-.02.502-.04.743-.04.32-.24.461-.54.461h-6.676c.363 1.193 1.296 1.666 2.338 1.666.56 0 1.12-.2 1.48-.501.2-.161.36-.261.62-.261h1.919c.36 0 .56.22.4.541-.78 1.747-2.42 2.65-4.459 2.65-3.319 0-5.438-2.308-5.438-5.3 0-2.99 2.199-5.298 5.358-5.298 2.92 0 5.038 2.268 5.038 5.299Zm-11.515-5.18c.34-.06.52.161.52.483v2.368c0 .321-.14.482-.4.482-1.12-.04-2.199.522-2.199 2.128v4.235c0 .322-.18.502-.5.502h-1.999c-.32 0-.5-.18-.5-.502V8.19c0-.32.18-.502.5-.502h1.999c.32 0 .5.181.5.502v1.085c.298-1.007 1.086-1.597 2.08-1.747Zm-6.896 0c.34-.06.52.161.52.483v2.368c0 .321-.14.482-.4.482-1.12-.04-2.2.522-2.2 2.128v4.235c0 .322-.18.502-.5.502h-1.998c-.32 0-.5-.18-.5-.502V8.19c0-.32.18-.502.5-.502H60.4c.32 0 .5.181.5.502v1.085c.298-1.007 1.086-1.597 2.079-1.747Zm-44.28 7.829a2.383 2.383 0 0 1-2.379 2.387.5.5 0 0 1-.499-.5V6.758a.5.5 0 0 1 .5-.501h1.878a.5.5 0 0 1 .5.5v8.597Zm-5.294 2.387h-1.88a.5.5 0 0 1-.498-.5V9.63a.5.5 0 0 1 .499-.501h1.879a.5.5 0 0 1 .499.5v7.612a.5.5 0 0 1-.5.501Zm-4.795 0H6.732a.5.5 0 0 1-.5-.5v-9.04a.5.5 0 0 1 .5-.5H8.61a.5.5 0 0 1 .5.5v9.04a.5.5 0 0 1-.5.5ZM15.821 0H9.11C4.08 0 0 4.095 0 9.146V23.5A.5.5 0 0 0 .499 24H15.82c5.031 0 9.11-4.095 9.11-9.146V9.146C24.93 4.095 20.852 0 15.82 0Z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </header>
</div>
`;
